### PR TITLE
Refactor miner

### PR DESCRIPTION
**Pull request by illya13**

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -182,122 +164,41 @@ func (app *EthermintApplication) validateTx(tx *ethTypes.Transaction) error {
...
-
-	//	app.backend.Ethereum().TxPool().RemoveBatch(app.currentTransactions)
-
-	// return validator updates
-	if app.strategy != nil {
-		return abciTypes.ResponseEndBlock{Diffs: app.strategy.GetUpdatedValidators()}
-	}
-	return abciTypes.ResponseEndBlock{}
+	app.backend.AccumulateRewards(app.strategy)
+	return app.GetUpdatedValidators()
```
Comment (jaekwon) [inline app/app.go:192]: The naming here doesn't seem ideal given what it returns.  minor point.

**Review by jaekwon — APPROVED** [review]
Beautiful.

**Comment by ebuchman** [comment]
Very nice. Minimal changes that we can hopefully get merged upstream. The tx pool is still redundant and probably a liability - but at least we should be able to preserve order through it now. I wonder if there's a way to get an interface to the thing handling the web3 transaction endpoints so we can hijack and make the HTTP request to tendermint, rather than waiting to catch a PreEvent from the eth txpool.

**Comment by illya13** [comment]
`tx-pool` is still doing some good job like:
- allowing to accept `tx` out of order per `address`
- wait here 
- and only then deliver it to the `listeners`

But IMO it should be redesigned inside `geth` as well

**Pull request merged**